File: backend/app/models/vehicle.py
from pydantic import BaseModel, Field


class Vehicle(BaseModel):
    make: str = Field(..., description="Manufacturer (e.g. Toyota)")
    model: str = Field(..., description="Model name (e.g. Camry)")
    year: int = Field(..., description="Model year")
    trim: str | None = Field(default=None, description="Trim level (e.g. SE, XLE)")
    body_style: str | None = Field(
        default=None,
        description="Body style (sedan, SUV, truck, coupe, hatchback, van, wagon)",
    )
    color: str | None = Field(default=None, description="Exterior color")
    confidence: float = Field(
        default=1.0,
        ge=0.0,
        le=1.0,
        description="AI confidence in identification (0.0–1.0)",
    )

File: CarCrashAI/Models/Vehicle.swift
import Foundation

struct Vehicle: Codable, Equatable {
    let make: String
    let model: String
    let year: Int
    let bodyStyle: String?
    let color: String?
    let confidence: Double

    enum CodingKeys: String, CodingKey {
        case make, model, year, color, confidence
        case bodyStyle = "body_style"
    }

    init(make: String, model: String, year: Int, bodyStyle: String? = nil, color: String? = nil, confidence: Double = 1.0) {
        self.make = make
        self.model = model
        self.year = year
        self.bodyStyle = bodyStyle
        self.color = color
        self.confidence = min(max(confidence, 0.0), 1.0)
    }
}
